templates/.github/workflows/ci.yml

```diff
@@ -0,0 +1,135 @@
+# .github/workflows/prod-pipeline.yml
+# This is a github action pipeline for production.
+name: ci pipeline
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: ['*']
```

Comment:
Just for testing?

Reply:
no, so basically the build step has a condition to only run on `master`, meaning the `build step` will be skipped on pull-requests
https://github.com/commitdev/zero-deployable-react-frontend/blob/eeb8a270ae911800c8b624c176c02f6695654dfa/templates/.github/workflows/ci.yml#L25-L28

Otherwise we'll have to copy and paste the unit test twice into a different workflow, which may end up a better idea since It will be clearer, but the code will be exactly the same

Reply:
Yeah, might be better to split it into 2 workflows, one to run tests on branches, the other to do test + build + deploy on main.
Also can you change this conditional to be `master` or `main`?